.github/workflows/build-and-test-arm.yml

```diff
@@ -46,14 +46,14 @@ jobs:
     runs-on: ubuntu-22.04-arm
     steps:
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
-      - uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 # v5
+      - uses: actions/setup-go@0aaccfd150d50ccaeb58ebd88d36e91967a5f35b # v5
         with:
-          go-version: "~1.23.7"
+          go-version: "~1.24.0"
```

Comment:
Assume https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38726 still holds, we cannot use go1.24